docs/src/main/tut/docs/stack/README.md

```diff
@@ -0,0 +1,260 @@
+    val readerM: rd.ReaderM[F]
+    val writerM: wr.WriterM[F]
+  }
+}
+```
+
+To validate the order we check if the number of crates that is ordered bigger is than zero and that the requested apple cultivar is selled by the company.
+
+For the second part we use the `reader` effect algebra to read the set of apple cultivars from our `Config`.
+
+```tut:book
+import cats.implicits._
+
+import modules._
+import cats.data.{NonEmptyList, ValidatedNel}
+
+def validateOrder[F[_]](order: Order, customer: Customer)(implicit app: App[F]): FreeS.Par[F, ValidatedNel[String, Unit]] =
+  app.readerM.reader { config =>
+    val v = ().validNel[String]
+    v.ensure(NonEmptyList.of(
+      "Number of crates ordered should be bigger than zero"))(
+      _ => order.crates > 0) |+|
+    v.ensure(NonEmptyList.of(
+      "Apple cultivar is not available"))(
+      _ => config.cultivars.contains(order.cultivar.toLowerCase))
+  }
+```
+
+
+We will create a couple of domain specific exception case classes.
+
+```tut:book
+sealed abstract class AppleException(val message: String) extends Exception(message)
+case class CustomerNotFound(id: CustomerId)               extends AppleException(s"Customer $id can not be found")
+case class QuantityNotAvailable(error: String)            extends AppleException(error)
+case class ValidationFailed(errors: NonEmptyList[String]) extends AppleException(errors.intercalate("\n"))
+```
+
+We can use the `validateOrder` method in combination with our persistence algebras and the `error` and `writer` effect algebras, to create the `processOrder` method tying everything together.
+
+```tut:book
+def processOrder[F[_]](order: Order)(implicit app: App[F]): FreeS[F, String] =
+  for {
+    customerOpt <- app.persistence.customer.getCustomer(order.customerId)
+    customer    <- app.errorM.either(customerOpt.toRight(CustomerNotFound(order.customerId)))
+    _           <- app.writerM.tell(Vector(s"customer name is ${customer.name}"))
+    validation  <- validateOrder[F](order, customer)
+    _           <- app.errorM.either(validation.toEither.leftMap(ValidationFailed))
+    nbAvailable <- app.persistence.stock.checkQuantityAvailable(order.cultivar)
+    _           <- app.writerM.tell(Vector(s"availble crates of ${order.cultivar} apples"))
+    _           <- app.errorM.either(
+                     Either.cond(
+                       order.crates <= nbAvailable,
+                       (),
+                       QuantityNotAvailable(
+                         s"""There are not sufficient crates of ${order.cultivar} apples in stock
+                            |(only $nbAvailable available, while ${order.crates} needed).""".stripMargin)
+                     )
+                   )
+    _          <- app.persistence.stock.registerOrder(order)
+  } yield s"Order registered for customer ${order.customerId}"
+```
+
+As target type of our computation we choose a combination of `Kleisli` (also known as `ReaderT`), `WriterT` and `fs2.Task`.
```

Comment:
We should briefly explain why we choose this stack and what each one of the main monads stacked here resolve from the point of view of application concerns.